src/Microsoft.IdentityModel.Tokens/RsaCryptoServiceProviderProxy.cs

```diff
@@ -64,13 +65,18 @@ public class RSACryptoServiceProviderProxy : RSA
         /// </summary>
         public override string KeyExchangeAlgorithm => _rsa.KeyExchangeAlgorithm;
 
+#pragma warning disable CS0162 // Disable warning on non-Windows platforms - CS0162: Unreachable code detected
         /// <summary>
         /// Initializes an new instance of <see cref="RSACryptoServiceProviderProxy"/>.
         /// </summary>
         /// <param name="rsa"><see cref="RSACryptoServiceProvider"/></param>
         /// <exception cref="ArgumentNullException">if <paramref name="rsa"/> is null.</exception>
+        /// <exception cref="PlatformNotSupportedException">RSACryptoServiceProviderProxy creation is only supported on Windows platform.</exception>
         public RSACryptoServiceProviderProxy(RSACryptoServiceProvider rsa)
         {
+#if !WINDOWS
+            throw new PlatformNotSupportedException(LogMessages.IDX10688);
```

Comment:
>PlatformNotSupportedException [](start = 22, length = 29)

this was what I would have considered you should throw in the adapter class...

Reply:
We don't have any code path in Wilson that would instantiate this class on non-Windows platforms. As RSACryptoServiceProviderProxy is a public class, my desire here was to explicitly state that this class is Windows-only.

Reply:
@brentschmaltz:
#if WINDOWS would fail a build as there is no RsaCryptoServiceProvider class in NETSTANDARD1_4.

Changing this line
`#if NET45 || NET451 || NET461 || NETSTANDARD2_0)`
to
`#if WINDOWS && (NET45 || NET451 || NET461 || NETSTANDARD2_0)` requires additional refactoring of AsymmetricAdapter class.

I tried to minimize the number of changes, but I'm open to further refactor AsymmetricAdapter and RsaCryptoServiceProvider in case that it's not preferable to throw a platform exception.

Reply:
About how much work would it be?

Reply:
too much work, just leave it as it is.